File: kafka-helper-domain/src/main/java/com/hg/kafka/helper/domain/entity/HostInfo.java
package com.hg.kafka.helper.domain.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;

import java.util.Date;

@Document(collection = "host_info")
@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class HostInfo {
    @Id
    private String id;
    @Field("host_name")
    private String hostName;
    @Field("hash_key")
    private String hashkey;
    @Field("status")
    private String status;
    @Field("updated_at")
    private Date updatedAt;
}

File: kafka-helper-domain/src/main/kotlin/com/hg/kafka/helper/domain/entity/HostInfo.kt
package com.hg.kafka.helper.domain.entity

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document
import org.springframework.data.mongodb.core.mapping.Field
import java.util.*

@Document("host_info")
data class HostInfo(
        @Id var id: String? = null,
        @Field("host_name") var hostName: String,
        @Field("hash_key") var hashkey: String,
        @Field("status") var status: String,
        @Field("updated_at") var updatedAt: Date
)